packages/react-native/React/CoreModules/RCTDeviceInfo.mm

```diff
@@ -74,16 +76,6 @@ - (void)initialize
                                            selector:@selector(interfaceFrameDidChange)
                                                name:UIDeviceOrientationDidChangeNotification
                                              object:nil];
-
-  // TODO T175901725 - Registering the RCTDeviceInfo module to the notification is a short-term fix to unblock 0.73
-  // The actual behavior should be that the module is properly registered in the TurboModule/Bridge infrastructure
-  // and the infrastructure imperatively invoke the `invalidate` method, rather than listening to a notification.
-  // This is a temporary workaround until we can investigate the issue better as there might be other modules in a
-  // similar situation.
-  [[NSNotificationCenter defaultCenter] addObserver:self
-                                           selector:@selector(invalidate)
-                                               name:RCTBridgeWillInvalidateModulesNotification
```

Comment:
We can't really remove this. 
If you look at the change that introduced this, it solves another bug: https://github.com/facebook/react-native/issues/42120.

Reply:
Also, the PR that introduced this is https://github.com/facebook/react-native/commit/d46d80d2ef6cdc05c5a1b2429ed1df93410ae36f and a further fix later in https://github.com/facebook/react-native/commit/91279a822be949051866a694148e509c1308035b